src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +46,83 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        LocalDate from = parseDate(dateFrom);
+        LocalDate to = parseDate(dateTo);
+        if (from.isAfter(to)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        Map<String, Integer> nameToEarnings = createNameToEarningsMap(names);
```

Comment:
map - is some kind of two-dimension array, but more complicated one. try to solve without it

Reply:
ok

Reply:
thank you. Even though it was cumbersome, the complexity of the previous version was n, now it is n^2.

Reply:
yeah, but what about HashMap methods complexity?

Reply:
Thank you!))